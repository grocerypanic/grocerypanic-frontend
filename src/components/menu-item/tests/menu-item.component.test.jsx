import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import MenuItem from "../menu-item.component";

const mockChoose = jest.fn();

const props = {
  name: "someName",
  location: "/someLocation",
  choose: mockChoose,
};

describe("Setup Environment", () => {
  let utils;

  beforeEach(() => {
    jest.clearAllMocks();
    utils = render(<MenuItem {...props} />);
  });

  afterEach(cleanup);

  it("should render with the correct message", () => {
    expect(utils.getByText(props.name)).toBeTruthy();
  });

  it("should handle a click correctly", () => {
    const node = utils.getByTestId("MenuElement");
    fireEvent.click(node, "click");
    expect(props.choose).toHaveBeenCalledWith(props.location);
  });

  it("should match the snapshot on file (styles)", () => {
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
