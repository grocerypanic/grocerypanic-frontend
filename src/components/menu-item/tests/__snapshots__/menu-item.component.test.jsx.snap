// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
  cursor: pointer;
}

<li
  class="c0"
>
  <div
    class="c1"
    data-testid="MenuElement"
  >
    someName
  </div>
</li>
`;
